### PR TITLE
Command Bus

### PR DESCRIPTION
**Pull request by jorge07**
This PR add a Command Bus pattern to manage the relation between use-cases and UI.

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -1196,6 +1196,132 @@
             "time": "2015-11-10 12:26:42"
         },
         {
+            "name": "league/tactician",
```
Comment (1ma) [inline composer.lock:1208]: New dependencies have been introduced but the composer.*json* file is the same.

fucky

File: app/AppKernel.php [hunk 2]
```diff
@@ -26,7 +26,10 @@ public function registerBundles()
             new Snc\RedisBundle\SncRedisBundle(),
             new Lexik\Bundle\JWTAuthenticationBundle\LexikJWTAuthenticationBundle(),
 
+            new League\Tactician\Bundle\TacticianBundle(),
+
             new Leos\UI\RestBundle\LeosUIRestBundle(),
+
```
Comment (1ma) [inline app/AppKernel.php]: Some men just want to watch the world burn.

File: src/Infrastructure/WalletBundle/Resources/config/application/use_case.yml [hunk 3]
```diff
@@ -1,5 +1,13 @@
...
+        class: Leos\Application\UseCase\Wallet\GetWalletHandler
         autowire: true
+        tags:
+          - { name: tactician.handler, command: Leos\Application\UseCase\Wallet\Request\GetWallet }
+
+    leos.use_case.find_wallet:
+        class: Leos\Application\UseCase\Wallet\FindWalletHandler
+        autowire: true
+        tags:
+          - { name: tactician.handler, command: Leos\Application\UseCase\Wallet\Request\Find }
```
Comment (1ma) [inline src/Infrastructure/WalletBundle/Resources/config/application/use_case.yml]: dat (missing) newline

**Review by 1ma — CHANGES_REQUESTED** [review]
needs correct composer.json

**Comment by 1ma** [comment]
Also, a couple of general questions:

- How does the Command Bus know which Handlers must execute which Commands? I have not seen any "mappings" of this kind anywhere, apart from the type hinting in the `handle` methods. If that's really enough for Tactician to figure out how should it work it is kinda cool.

- The PR doesn't add any middlewares, which, as far as I understand are the major motivator for using a Command Bus. So given this fact, how is this better than just injecting the handler(s) you need into each controller instead of the Command Bus?

**Comment by jorge07** [comment]
@1ma 
1 -> `src/Infrastructure/WalletBundle/Resources/config/application/use_case.yml` that's one of the files. Look at `src/Infrastructure/**/use_case.yml` files.

2 -> You're right, doesn't add any middleware, but add the possibility to do it. The idea was use `Doctrine Middleware` but when I read in his doc that: `Each command you execute will now open and close a new transaction.` I refused because I don't wanna open a connection to the database if I don't need it. So, basically abstracts the UI of the Domain system, and you just need to know the command you wanna run.

**Comment by 1ma** [comment]
LGTM now

**Pull request merged**